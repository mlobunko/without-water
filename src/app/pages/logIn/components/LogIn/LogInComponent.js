import React from 'react';
import Alert from 'shared/error/components/Alert';
import { FormGroup, Input, Label } from 'reactstrap';
import PropTypes from 'prop-types';

import { StyledForm } from './styles';

import Button from './Button';
import HeaderH1 from 'app/header/components/HeaderH1';
import Loader from 'shared/fetching/components/Loader';

export const LogInComponent = ({
  email,
  emailError,
  fetching,
  handleInputsChange,
  handleSubmit,
  isError,
  loginError,
  password,
  passwordError,
  onKeyPress,
  serverError
}) => (
  <React.Fragment>
    <HeaderH1 text={'Log In'} />
    <StyledForm
      onSubmit={handleSubmit}
      onKeyPress={fetching ? onKeyPress : undefined}
    >
      <FormGroup>
        <Label htmlFor="email">Email</Label>
        <Input
          id="email"
          placeholder="Enter email"
          type="text"
          value={email}
          autoComplete="on"
          onChange={handleInputsChange}
          invalid={!!loginError || !!emailError}
        />
        {!!emailError && <Alert color="danger" textError={emailError} />}
      </FormGroup>
      <FormGroup>
        <Label htmlFor="password">Password</Label>
        <Input
          id="password"
          placeholder="Enter password"
          type="password"
          value={password}
          autoComplete="on"
          onChange={handleInputsChange}
          invalid={!!loginError || !!passwordError}
        />
      </FormGroup>
      {fetching && <Loader login />}
      {isError || !!passwordError ? (
        <Alert
          color="danger"
          textError={loginError || serverError || passwordError}
        />
      ) : null}
      <Button
        text={'Log In'}
        outline
        color="dark"
        type="submit"
        disabled={!!fetching}
      />
    </StyledForm>
  </React.Fragment>
);

export default LogInComponent;

LogInComponent.propTypes = {
  email: PropTypes.string.isRequired,
  emailError: PropTypes.string.isRequired,
  fetching: PropTypes.bool.isRequired,
  handleInputsChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  isError: PropTypes.bool.isRequired,
  loginError: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  passwordError: PropTypes.string.isRequired,
  onKeyPress: PropTypes.func.isRequired,
  serverError: PropTypes.string.isRequired
};
