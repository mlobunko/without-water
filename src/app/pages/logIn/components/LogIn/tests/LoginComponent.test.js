import React from 'react';
import { shallow } from 'enzyme';

import { LogInComponent } from '../LogInComponent';

const initialProps = {
  email: 'fa@fa.com',
  emailError: '',
  fetching: false,
  handleInputsChange: jest.fn(),
  handleSubmit: jest.fn(),
  isError: false,
  loginError: '',
  password: '123456',
  passwordError: '',
  onKeyPress: jest.fn(),
  serverError: ''
};

describe('form', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('find handleSubmit', () => {
    const props = {
      ...initialProps
    };
    const wrapper = shallow(<LogInComponent {...props} />);
    expect(wrapper.find('StyledForm').prop('onSubmit')).toEqual(
      props.handleSubmit
    );
  });

  describe('onKeyPress', () => {
    it('property onKeyPress is onKeyPress function', () => {
      const props = {
        ...initialProps,
        fetching: true
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      expect(wrapper.find('StyledForm').prop('onKeyPress')).toEqual(
        props.onKeyPress
      );
    });

    it('property onKeyPress is undefined', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      expect(wrapper.find('StyledForm').prop('onKeyPress')).toEqual(undefined);
    });
  });

  describe('render', () => {
    it('renders correctly', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('input email', () => {
  it('onChange', () => {
    const props = {
      ...initialProps
    };
    const wrapper = shallow(<LogInComponent {...props} />);
    expect(wrapper.find('#email').prop('onChange')).toEqual(
      props.handleInputsChange
    );
  });

  describe('invalid', () => {
    describe('loginError is true', () => {
      const props = {
        ...initialProps,
        loginError: 'error'
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const input = wrapper.find('#email');
      it('find prop', () => {
        expect(input.prop('invalid')).toEqual(true);
      });
      it('renders correctly', () => {
        expect(input).toMatchSnapshot();
      });
    });

    describe('emailError is true', () => {
      const props = {
        ...initialProps,
        emailError: 'error'
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const input = wrapper.find('#email');

      it('find prop', () => {
        expect(input.prop('invalid')).toEqual(true);
      });
      it('renders correctly', () => {
        expect(input).toMatchSnapshot();
      });
    });

    describe('emailError, loginError are false', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const input = wrapper.find('#email');
      it('find prop', () => {
        expect(input.prop('invalid')).toEqual(false);
      });
      it('renders correctly', () => {
        expect(input).toMatchSnapshot();
      });
    });
  });
});

describe('input password', () => {
  it('onChange', () => {
    const props = {
      ...initialProps
    };
    const wrapper = shallow(<LogInComponent {...props} />);
    expect(wrapper.find('#password').prop('onChange')).toEqual(
      props.handleInputsChange
    );
  });

  describe('invalid', () => {
    describe('loginError is true', () => {
      const props = {
        ...initialProps,
        loginError: 'error'
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const input = wrapper.find('#password');
      it('find prop', () => {
        expect(input.prop('invalid')).toEqual(true);
      });
      it('renders correctly', () => {
        expect(input).toMatchSnapshot();
      });
    });

    describe('passwordError is true', () => {
      const props = {
        ...initialProps,
        passwordError: 'error'
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const input = wrapper.find('#password');

      it('find prop', () => {
        expect(input.prop('invalid')).toEqual(true);
      });
      it('renders correctly', () => {
        expect(input).toMatchSnapshot();
      });
    });

    describe('emailError, loginError are false', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const input = wrapper.find('#password');

      it('find prop', () => {
        expect(input.prop('invalid')).toEqual(false);
      });
      it('renders correctly', () => {
        expect(input).toMatchSnapshot();
      });
    });
  });
});

describe('alert under the email', () => {
  describe('render', () => {
    it('emailError is true', () => {
      const props = {
        ...initialProps,
        emailError: 'emailError'
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const alert = wrapper.find('[textError="emailError"]');
      expect(alert).toMatchSnapshot();
    });
    it('emailError is false', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      expect(wrapper.find('[color="danger"]')).toHaveLength(0);
    });
  });
});

describe('alert under the password', () => {
  describe('render', () => {
    describe('isError is true', () => {
      it('loginError', () => {
        const props = {
          ...initialProps,
          isError: true,
          loginError: 'loginError'
        };
        const wrapper = shallow(<LogInComponent {...props} />);
        const alert = wrapper.find('[textError="loginError"]');
        expect(alert).toMatchSnapshot();
      });

      it('serverError', () => {
        const props = {
          ...initialProps,
          isError: true,
          serverError: 'serverError'
        };
        const wrapper = shallow(<LogInComponent {...props} />);
        const alert = wrapper.find('[textError="serverError"]');
        expect(alert).toMatchSnapshot();
      });
    });

    it('passwordError is true', () => {
      const props = {
        ...initialProps,
        passwordError: 'passwordError'
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const alert = wrapper.find('[textError="passwordError"]');
      expect(alert).toMatchSnapshot();
    });

    it('passwordError, isError are false', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      expect(wrapper.find('[color="danger"]')).toHaveLength(0);
    });
  });
});

describe('submit button', () => {
  describe('render', () => {
    it('fetching is true', () => {
      const props = {
        ...initialProps,
        fetching: true
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const button = wrapper.find('[type="submit"]');
      expect(button).toMatchSnapshot();
    });

    it('fetching is false', () => {
      const props = {
        ...initialProps
      };
      const wrapper = shallow(<LogInComponent {...props} />);
      const button = wrapper.find('[type="submit"]');
      expect(button).toMatchSnapshot();
    });
  });
});
