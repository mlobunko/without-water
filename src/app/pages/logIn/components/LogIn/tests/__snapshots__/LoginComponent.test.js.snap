// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert under the email render emailError is true 1`] = `
<Alert
  color="danger"
  textError="emailError"
/>
`;

exports[`alert under the password render isError is true loginError 1`] = `
<Alert
  color="danger"
  textError="loginError"
/>
`;

exports[`alert under the password render isError is true serverError 1`] = `
<Alert
  color="danger"
  textError="serverError"
/>
`;

exports[`alert under the password render passwordError is true 1`] = `
<Alert
  color="danger"
  textError="passwordError"
/>
`;

exports[`form render renders correctly 1`] = `
<Fragment>
  <Component
    text="Log In"
  />
  <StyledForm
    onSubmit={[Function]}
  >
    <FormGroup
      tag="div"
    >
      <Label
        htmlFor="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email
      </Label>
      <Input
        autoComplete="on"
        id="email"
        invalid={false}
        onChange={[Function]}
        placeholder="Enter email"
        type="text"
        value="fa@fa.com"
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        htmlFor="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
      </Label>
      <Input
        autoComplete="on"
        id="password"
        invalid={false}
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value="123456"
      />
    </FormGroup>
    <Component
      color="dark"
      disabled={false}
      outline={true}
      text="Log In"
      type="submit"
    />
  </StyledForm>
</Fragment>
`;

exports[`input email invalid emailError is true renders correctly 1`] = `
<Input
  autoComplete="on"
  id="email"
  invalid={true}
  onChange={[Function]}
  placeholder="Enter email"
  type="text"
  value="fa@fa.com"
/>
`;

exports[`input email invalid emailError, loginError are false renders correctly 1`] = `
<Input
  autoComplete="on"
  id="email"
  invalid={false}
  onChange={[Function]}
  placeholder="Enter email"
  type="text"
  value="fa@fa.com"
/>
`;

exports[`input email invalid loginError is true renders correctly 1`] = `
<Input
  autoComplete="on"
  id="email"
  invalid={true}
  onChange={[Function]}
  placeholder="Enter email"
  type="text"
  value="fa@fa.com"
/>
`;

exports[`input password invalid emailError, loginError are false renders correctly 1`] = `
<Input
  autoComplete="on"
  id="password"
  invalid={false}
  onChange={[Function]}
  placeholder="Enter password"
  type="password"
  value="123456"
/>
`;

exports[`input password invalid loginError is true renders correctly 1`] = `
<Input
  autoComplete="on"
  id="password"
  invalid={true}
  onChange={[Function]}
  placeholder="Enter password"
  type="password"
  value="123456"
/>
`;

exports[`input password invalid passwordError is true renders correctly 1`] = `
<Input
  autoComplete="on"
  id="password"
  invalid={true}
  onChange={[Function]}
  placeholder="Enter password"
  type="password"
  value="123456"
/>
`;

exports[`submit button render fetching is false 1`] = `
<Component
  color="dark"
  disabled={false}
  outline={true}
  text="Log In"
  type="submit"
/>
`;

exports[`submit button render fetching is true 1`] = `
<Component
  color="dark"
  disabled={true}
  outline={true}
  text="Log In"
  type="submit"
/>
`;
